docs/testplan-style.md

```diff
@@ -0,0 +1,243 @@
+*   ...
+```
+
+## Development Approach for Tests
+
+### Vendor Neutrality
+
+Test plans should use a vendor-neutral mechanism to configure the DUT, e.g. in
+the OpenConfig public model or other OpenConfig protocols such as gNMI, gRIBI,
+and gNOI. It should depend only on the behavior of the open specification.
+
+### Device/Topology Neutrality
+
+Tests are to be built against a minimal topology that consists of a single
+device (referred to as DUT) against a reference implementation (referred to as
+an ATE) of routing protocols and data plane functionality.
+
+Each test plan must choose from one of the following testbeds and adhere to it
+for the entirety of the test plan:
+
+*   atedut\_2: two interconnected ports between ATE and DUT.
+*   atedut\_4: four interconnected ports between ATE and DUT.
+*   atedut\_12: 4 singleton ports and 8 breakout ports interconnected between
+    ATE and DUT. Ondatra testbed models the logical ports, but the test is
+    responsible for activating the breakout.
+*   dutdut: four interconnected ports between two DUTs.
+
+Test plans must not hardcode assumptions about the presence of specific hardware
+components (e.g. switch chip, controller card, line card). If needed, it may
+accept a "device profile" which parameterizes the expectations about the device.
+If the test plan is about a feature that does not make sense for a given device
+profile, the test should be skipped. For example, a test about controller card
+switchover should be skipped if the device profile indicates that the device is
+expected to have only one controller card, rather than skipping based on the
+number of controller cards detected from the device.
+
+### Functional Testing vs. Fuzzing
+
+Initially, the test plan will consist of manually identified functional features
+as opposed to automated parameter space coverage known as fuzzing. We are
+prioritizing on validating device functionality (e.g. establishing a BGP session
+based on a specific input BGP configuration) that are critical to the device
+operation, rather than compliance solely based on the input schema (e.g.
+enumerating that all uint32 can be used as BGP AS number).
+
+### Telemetry as a Component of a Test
+
+In some previous testing implementations, specific tests have been designed to
+specifically cover the correct reporting of telemetry. This approach tends to
+duplicate test setup across functional tests and telemetry-specific tests. We
+choose to implement tests that cover telemetry testing alongside the
+functionality testing. Generally, telemetry from both the DUT and ATE should be
+used as pass/fail criteria of the test, such that telemetry is validated as a
+core component of each test.
+
+## Readable Language
+
+All code and documentation should follow
+[Google developer documentation style guide](https://developers.google.com/style/word-list)
+for the use of inclusive language.
+
+Test plan should be written in a natural language. Avoid introducing your own
```

Comment:
Can you explain what this means? In some tests it is hard to define things solely in prose, and having some form of parseable pseudo-code seems quite useful. It'd be good to understand the intention behind this statement.